### PR TITLE
LPS-165679 [FE] Update index selector to use search index API and save indexConfiguration

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-165679

**On the blueprints edit page:**
- Uses the new search index API implemented in https://issues.liferay.com/browse/LPS-163750 to populate the search indexes selector. 
- Saves the selected index name to the blueprint configuration in the format `{indexConfiguration: {indexName: '${name}'}}`

Screen Shot 2022-10-14 at 8 52 46 AM

### CODE REVIEW

**Comment by thektan** [comment]
ci:test:relevant

**Comment by thektan** [comment]
ci:stop

**Comment by thektan** [comment]
Fixing commit message

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:relevant

**Comment by thektan** [comment]
Seems like I need to wrap more of the frontend logic in a feature flag and add better error handling. Making these fixes.

**Comment by thektan** [comment]
ci:test:search

**Comment by thektan** [comment]
ci:test:search

**Comment by thektan** [comment]
@oliv-yu Pushed up schema name changes noted in https://github.com/liferay-search/liferay-portal/pull/599 and ready to review/forward!

**Comment by thektan** [comment]
@oliv-yu Good to review ✅

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:search

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js [hunk 1]
```diff
@@ -477,8 +508,8 @@ function EditSXPBlueprintForm({
 			sortConfiguration: sortConfig ? JSON.parse(sortConfig) : {},
 		};
 
-		if (indexConfig) {
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js:480]: The acceptance criteria says that if company index was being used, it'd default to empty. Was wondering if that meant defining `indexConfiguration` at all? Also would the default value be `{indexName: ''}`? Though I'm not sure if it'd ever reach that part.

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js [hunk 2]
```diff
@@ -477,8 +508,8 @@ function EditSXPBlueprintForm({
...
-			configuration.indexConfiguration = indexConfig;
+		if (featureFlagLps153813) {
+			configuration.indexConfiguration = indexConfig || {};
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js:512]: The acceptance criteria says that if company index was being used, it'd default to empty. Was wondering if that meant defining `indexConfiguration` at all? Also would the default value be `{indexName: ''}`? Though I'm not sure if it'd ever reach that part.

Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js:512]: Yes the default is set to an empty value `{indexName: ''}` here:
https://github.com/liferay-search/liferay-portal/blob/45bdc16ab5af8e730687b135503645b830a338df/modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js#L400-L402

Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js:512]: And when a user selects the "Company Index" selection this is where the empty value is set as well.
https://github.com/liferay-search/liferay-portal/blob/45bdc16ab5af8e730687b135503645b830a338df/modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/configuration_tab/index.js#L58

**Comment by thektan** [comment]
ci:test:search

**Review by oliv-yu — APPROVED** [review]
looks great! thanks for taking the time to add documentation and check about wording! 💯

**Comment by thektan** [comment]
ci:forward

**Pull request closed without merging**